src/plots/cartesian/axes.js

```diff
@@ -2966,7 +2966,11 @@ axes.drawLabels = function(gd, ax, opts) {
     var axId = ax._id;
     var axLetter = axId.charAt(0);
     var cls = opts.cls || axId + 'tick';
-    var vals = opts.vals;
+
+    var jump = ax.ticklabeljump;
+    var vals = jump ?
+        opts.vals.filter(function(_, i) { return i % (jump + 1) === 0; }) :
```

Comment:
If I'm understanding this correctly, this way we're always going to get the first label that appears in the range, then every nth after that. But what we really want is to always get the label corresponding to `tick0` if that's within the range, otherwise start n ticks away from `tick0`. An important consequence of this is that as you pan around on a graph the same label will remain displayed or not displayed as long as it's in range.

Reply:
Certain filtering logic is now moved from `drawLabels` to `calcTicks` by https://github.com/plotly/plotly.js/pull/6088/commits/29ab44421b74dcac8ed8f69a6066c1fd120a95df.
Next I'd try to improve on that to keep the same labels around...

Reply:
This is now addressed by https://github.com/plotly/plotly.js/pull/6088/commits/3b259ed3fc7fd9b992ac20d103c74dd4e246ba48.